src/sagemaker_training/mpi.py

```diff
@@ -177,6 +179,23 @@ def _orted_process():  # pylint: disable=inconsistent-return-statements
         time.sleep(1)
 
 
+def _smddp_command(instance_type):  # type: (str) -> list[str]
+    """When a task is of modelparallel and ddp_dist_backend is auto,
+    we use smddpmprun to set up necessary environment variables if possible.
+    """
+    command = []
+    env = environment.Environment()
+    if env.is_modelparallel_enabled:
+        mp_parameters = json.loads(os.environ.get(params.SM_HP_MP_PARAMETERS, "{}"))
+        ddp_dist_backend = mp_parameters.get("ddp_dist_backend", "auto")
+        if ddp_dist_backend == "auto":
+            if env.is_smddp_smddpmprun_installed:
+                command.extend(["smddpmprun", instance_type, "--allow-bypass"])
+        else:
+            logger.info("NCCL backend is used as specified")
```

Comment:
can we parameterize the logger info here? i.e., use like below
`logger.info(f"{ddp_dist_backend} is used as backend for training") `

Reply:
Sure, will update it.